webapp/ruby/app.rb

```diff
@@ -0,0 +1,544 @@
+    end
+  end
+
+  post '/initialize' do
+    unless system('../mysql/db/init.sh')
+      logger.error('Initialize script error')
+      halt 500
+    end
+
+    { language: 'ruby' }.to_json
+  end
+
+  get '/api/chair/low_priced' do
+    sql = "SELECT * FROM chair WHERE stock > 0 ORDER BY price ASC, id ASC LIMIT #{LIMIT}" # XXX:
+    chairs = db.query(sql).to_a
+    { chairs: chairs }.to_json
+  end
+
+  get '/api/chair/search' do
+    search_queries = []
+    query_params = []
+
+    if params[:priceRangeId] && params[:priceRangeId].size > 0
+      chair_price = CHAIR_SEARCH_CONDITION[:price][:ranges][params[:priceRangeId].to_i]
+      unless chair_price
+        logger.error "priceRangeID invalid: #{params[:priceRangeId]}"
+        halt 400
+      end
+
+      if chair_price[:min] != -1
+        search_queries << 'price >= ?'
+        query_params << chair_price[:min]
+      end
+
+      if chair_price[:max] != -1
+        search_queries << 'price < ?'
+        query_params << chair_price[:max]
+      end
+    end
+
+    if params[:heightRangeId] && params[:heightRangeId].size > 0
+      chair_height = CHAIR_SEARCH_CONDITION[:height][:ranges][params[:heightRangeId].to_i]
+      unless chair_height
+        logger.error "heightRangeId invalid: #{params[:heightRangeId]}"
+        halt 400
+      end
+
+      if chair_height[:min] != -1
+        search_queries << 'height >= ?'
+        query_params << chair_height[:min]
+      end
+
+      if chair_height[:max] != -1
+        search_queries << 'height < ?'
+        query_params << chair_height[:max]
+      end
+    end
+
+    if params[:widthRangeId] && params[:widthRangeId].size > 0
+      chair_width = CHAIR_SEARCH_CONDITION[:width][:ranges][params[:widthRangeId].to_i]
+      unless chair_width
+        logger.error "widthRangeId invalid: #{params[:widthRangeId]}"
+        halt 400
+      end
+
+      if chair_width[:min] != -1
+        search_queries << 'width >= ?'
+        query_params << chair_width[:min]
+      end
+
+      if chair_width[:max] != -1
+        search_queries << 'width < ?'
+        query_params << chair_width[:max]
+      end
+    end
+
+    if params[:depthRangeId] && params[:depthRangeId].size > 0
+      chair_depth = CHAIR_SEARCH_CONDITION[:depth][:ranges][params[:depthRangeId].to_i]
+      unless chair_depth
+        logger.error "depthRangeId invalid: #{params[:depthRangeId]}"
+        halt 400
+      end
+
+      if chair_depth[:min] != -1
+        search_queries << 'depth >= ?'
+        query_params << chair_depth[:min]
+      end
+
+      if chair_depth[:max] != -1
+        search_queries << 'depth < ?'
+        query_params << chair_depth[:max]
+      end
+    end
+
+    if params[:kind] && params[:kind].size > 0
+      search_queries << 'kind = ?'
+      query_params << params[:kind]
+    end
+
+    if params[:color] && params[:color].size > 0
+      search_queries << 'color = ?'
+      query_params << params[:kind]
```

Comment:
検索条件に color を含む検索にて、結果が 0 件になっていました。
おそらくここが原因かと思われます。

```suggestion
      query_params << params[:color]
```